### ISSUE TITLE
Inline assembler problem

### ISSUE THREAD

**Issue by peterhinch** [micropython/micropython#1099]
The following function produces an inline assembler error on loading the module. If the bne line is commented out the problem disappears. Other variants such as bpl, beq etc. are also affected.
```python
@micropython.asm_thumb
def foo(r0):
    cmp(r0, 0)
    bne(BAR)
    mov(r0, 0)
    lbl(BAR)
```

**Comment by danicampora** [comment]
The inline assembly error is gone if you declare the label as: `label(BAR)`, .i.e:
```asm
@micropython.asm_thumb
def foo(r0):
    cmp(r0, 0)
    bne(BAR)
    mov(r0, 0)
    label(BAR)
```
AFAIK, according to http://micropython.org/doc/tut-asm that's the way of declaring labels ;-)

**Comment by dpgeorge** [comment]
The bug is that it is not reporting that the call "lbl" is an undefined instruction.

**Comment by danicampora** [comment]
True, that's what @peterhinch reported in #1100

**Comment by peterhinch** [comment]
It's perhaps worth noting that other syntax errors result in a message lacking a line number. For example
```
mov(r7, [r4, 0])
```
(valid opcode, invalid arguments for that instruction)

**Comment by dpgeorge** [comment]
Now fully fixed by 8dfbd2d5894c98bd7b48e3e69bec0216d06ebb47 and 0d967b8ae4f26815549aa2fb301e7bbaa5e262c2.  One issue was that any unknown instruction with 1 argument was being assembled as cpsid, which disables interrupts.
